### PR TITLE
Documenting changes that resulted from issue 1475

### PR DESCRIPTION
**Pull request by Buzzardo**
Scott Frederick asked me to document the changes he made while implementing the changes needed for issue 1475 (https://github.com/spring-cloud/spring-cloud-config/issues/1475). This PR adds the details that Scott explained to me. Thanks, Scott. :)

### CODE REVIEW

File: docs/src/main/asciidoc/spring-cloud-config.adoc [hunk 1]
```diff
@@ -663,6 +663,14 @@ You should see a response similar to the following:
 }
 ----
 
+The default way for a client to provide the necessary authentication to let Config Server talk to Vault is to set the X-Config-Token header.
+However, you can instead omit the header and configure the authentication in the server, by setting the same configuration properties as Spring Cloud Vault.
+The property to set is `spring.cloud.config.server.vault.authentication`.
+See the https://cloud.spring.io/spring-cloud-vault/reference/html/#vault.config.authentication[Spring Cloud Vault Reference Guide] for more detail.
+
+IMPORTANT: If you omit the X-Config-Token header and use a server property to set the authentication, the Config Server application needs an additional dependency on Spring Vault (not Spring Cloud Vault) to enable the additional authentication options.
```
Comment (scottfrederick) [inline docs/src/main/asciidoc/spring-cloud-config.adoc]: I think we can leave out the `(not Spring Cloud Vault)` part

File: docs/src/main/asciidoc/spring-cloud-config.adoc [hunk 2]
```diff
@@ -663,6 +663,14 @@ You should see a response similar to the following:
 }
 ----
 
+The default way for a client to provide the necessary authentication to let Config Server talk to Vault is to set the X-Config-Token header.
+However, you can instead omit the header and configure the authentication in the server, by setting the same configuration properties as Spring Cloud Vault.
+The property to set is `spring.cloud.config.server.vault.authentication`.
```
Comment (scottfrederick) [inline docs/src/main/asciidoc/spring-cloud-config.adoc]: `spring.cloud.config.server.vault.authentication` should be set to one of the supported Authentication Methods. Other properties specific to the chosen authentication method may also need to be set, using the same property names as documented for `spring.cloud.vault` but using the `spring.cloud.config.server.vault` prefix.

File: docs/src/main/asciidoc/spring-cloud-config.adoc (hunk 1 above)
Comment (Buzzardo) [reply docs/src/main/asciidoc/spring-cloud-config.adoc]: OK

File: docs/src/main/asciidoc/spring-cloud-config.adoc (hunk 2 above)
Comment (Buzzardo) [reply docs/src/main/asciidoc/spring-cloud-config.adoc]: Good detail. Thanks. :)

**Pull request merged**